src/read_concern.ts

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * The **ReadConcern** class is a class that represents a MongoDB ReadConcern.
+ *
+ * @class
+ * @property {string} level The read concern level
+ * @see https://docs.mongodb.com/manual/reference/read-concern/index.html
+ */
+class ReadConcern {
+  level?: string
+
+  /**
+   * Constructs a ReadConcern from the read concern properties.
+   *
+   * @param {string} [level] The read concern level ({'local'|'available'|'majority'|'linearizable'|'snapshot'})
+   */
+  constructor(level: string) {
+    if (level != null) {
+      this.level = level;
+    }
+  }
+
+  /**
+   * Construct a ReadConcern given an options object.
+   *
+   * @param {object} options The options object from which to extract the write concern.
+   * @returns {ReadConcern}
+   */
+  static fromOptions(options: any) {
+    if (options == null) {
+      return;
+    }
+
+    if (options.readConcern) {
+      if (options.readConcern instanceof ReadConcern) {
+        return options.readConcern;
+      }
+
+      return new ReadConcern(options.readConcern.level);
+    }
+
+    if (options.level) {
+      return new ReadConcern(options.level);
+    }
+  }
+
+  static get MAJORITY() {
+    return 'majority';
+  }
+
+  static get AVAILABLE() {
+    return 'available';
+  }
+
+  static get LINEARIZABLE() {
+    return 'linearizable';
+  }
+
+  static get SNAPSHOT() {
+    return 'snapshot';
+  }
+}
+
+module.exports = ReadConcern;
```

Comment:
(Just a question, for discussion sake) We don't wanna use es6 `export` here yet cus it does emit some extra code but do we forsee a time when we're fine with that extra code? (maybe when the whole code base is ts, we would consider migrating then?)

The extra code:
```js
Object.defineProperty(exports, "__esModule", { value: true })
```

Reply:
If the guiding principle is that we are using TypeScript for typing, not adding language features (transpiling), then I think we would want to support es6 modules when node does. That means we couldn't support es6 modules until we set a minimum supported version of Node 13 (I believe). 

I'm trying to better understand the use case, is it just more ergonomic to use `import` and `export` declarations? Or are you perceiving some other potential issues we might run into?

Reply:
I think we'll still have type definition emitting problems unless tsc makes the fix, or we rely on es6 module syntax. So we can just wait to see if that is fixed when we get to an all ts state.